workos/exceptions.py

```diff
@@ -8,70 +8,35 @@ class BaseRequestException(Exception):
     def __init__(
         self,
         response: httpx.Response,
-        message: Optional[str] = None,
-        error: Optional[str] = None,
-        errors: Optional[Mapping[str, Any]] = None,
-        error_description: Optional[str] = None,
-        code: Optional[str] = None,
-        pending_authentication_token: Optional[str] = None,
+        response_json: Optional[Mapping[str, Any]],
     ) -> None:
-        super(BaseRequestException, self).__init__(message)
+        super(BaseRequestException, self).__init__(response_json)
 
-        self.message = message
-        self.error = error
-        self.errors = errors
-        self.code = code
-        self.error_description = error_description
-        self.pending_authentication_token = pending_authentication_token
-        self.extract_and_set_response_related_data(response)
-
-    def extract_and_set_response_related_data(self, response: httpx.Response) -> None:
         self.response = response
+        self.response_json = response_json
 
-        try:
-            response_json = response.json()
-            self.message = response_json.get("message")
-            self.error = response_json.get("error")
-            self.errors = response_json.get("errors")
-            self.code = response_json.get("code")
-            self.error_description = response_json.get("error_description")
-            self.pending_authentication_token = response_json.get(
-                "pending_authentication_token"
-            )
-        except ValueError:
-            self.message = None
-            self.error = None
-            self.errors = None
-            self.code = None
-            self.error_description = None
-            self.pending_authentication_token = None
-
-        headers = response.headers
-        self.request_id = headers.get("X-Request-ID")
-
-    def __str__(self) -> str:
-        message = self.message or "No message"
-        exception = "(message=%s" % message
+        self.message = self.extractFromJson("message", "No message")
```

Comment:
This is 100x nicer than `except ValueError`, good stuff

Reply:
Thanks! Also lol snake case